pkg/solana/chainwriter/ata_creation.go

```diff
@@ -100,67 +98,34 @@ func CreateATAs(ctx context.Context, args any, lookups []ATALookup, derivedTable
 	return createATAInstructions, nil
 }
 
-func (s *SolanaChainWriterService) handleATACreation(ctx context.Context, createATAinstructions []solana.Instruction, methodConfig MethodConfig, contractName, method string, feePayer solana.PublicKey) error {
+func (s *SolanaChainWriterService) handleATACreation(ctx context.Context, createATAinstructions []solana.Instruction, methodConfig MethodConfig, contractName, method string, feePayer solana.PublicKey) (string, error) {
 	if len(createATAinstructions) == 0 {
-		return nil
+		return "", nil
 	}
 	blockhash, err := s.client.LatestBlockhash(ctx)
 	if err != nil {
-		return fmt.Errorf("error fetching latest blockhash: %w", err)
+		return "", fmt.Errorf("error fetching latest blockhash: %w", err)
 	}
 
 	if len(createATAinstructions) > maxAtas {
-		return fmt.Errorf("too many ATAs to create: %d, max allowed: %d", len(createATAinstructions), maxAtas)
+		return "", fmt.Errorf("too many ATAs to create: %d, max allowed: %d", len(createATAinstructions), maxAtas)
 	}
 	ataTx, ataErr := solana.NewTransaction(
 		createATAinstructions,
 		blockhash.Value.Blockhash,
 		solana.TransactionPayer(feePayer),
 	)
 	if ataErr != nil {
-		return fmt.Errorf("error constructing ATA transaction: %w", err)
+		return "", fmt.Errorf("error constructing ATA transaction: %w", err)
 	}
 	ataUUID := fmt.Sprintf("ATA-%s", uuid.NewString())
 
 	s.lggr.Infow("Sending create ATA transaction", "contract", contractName, "method", method)
 
 	// Enqueue ATA transaction
 	if err = s.txm.Enqueue(ctx, methodConfig.FromAddress, ataTx, &ataUUID, blockhash.Value.LastValidBlockHeight); err != nil {
```

Comment:
Maybe we can double check with CCIP but think we may want to enable compute unit limit estimation for ATA transactions. The default limit might be larger than what we need for the ATA tx.

Reply:
Hmm, how would you imagine this working? Would they need to provide a separate limit? I can start a thread with CCIP

Reply:
If they want to explicitly cap the cost of this, I guess that's something they would need to provide. Don't think that's feasible though so I was just thinking we do our best effort to keep costs low and just set this override for this tx. That way we're not overspending on the CU limit for the ATA tx.
```suggestion
	if err = s.txm.Enqueue(ctx, methodConfig.FromAddress, ataTx, &ataUUID, blockhash.Value.LastValidBlockHeight, txmutils.SetEstimateComputeUnitLimit(true)); err != nil {
```